lib/debug_logging/class_notifier.rb

```diff
@@ -20,29 +21,39 @@ def notifies(*methods_to_notify)
             config_proxy = if (proxy = instance_variable_get(DebugLogging::Configuration.config_pointer('k', method_to_notify)))
                              proxy
                            else
-                             proxy = if !payload.empty?
-                                       Configuration.new(**debug_config.to_hash.merge(payload))
+                             proxy = if !config_opts.empty?
+                                       DebugLogging::Configuration.new(**debug_config.to_hash.merge(config_opts))
                                      else
                                        debug_config
                                      end
                              proxy.register(method_to_notify)
                              instance_variable_set(DebugLogging::Configuration.config_pointer('k', method_to_notify), proxy)
```

Comment:
Caches the config for this method.  Because of the cache check, this ensures that the `subscribe` below only runs the very first time a method is invoked, and that it runs before the `instrumentation` runs.